analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd

```diff
@@ -0,0 +1,290 @@
+
+```
+
+### LGG, IDH
+
+This subtype harbors an IDH R132 mutation
+```{r}
+idh_biospecimen <- lgat_snv %>%
+  filter(IDH_mut=="Yes") %>%
+  mutate(molecular_subtype = "LGG, IDH") %>%
+  select("Kids_First_Biospecimen_ID","molecular_subtype") 
+
+```
+
+### LGG, H3
+
+This subtype harbors an H3F3A K28M or G35R/V mutation
+```{r}
+h3_biospecimen <- lgat_snv %>%
+  filter(H3.1_mut =="Yes"| H3.2_mut =="Yes" |H3.3_mut =="Yes") %>%
+  mutate(molecular_subtype = "LGG, H3") %>%
+  select("Kids_First_Biospecimen_ID","molecular_subtype") 
+
+```
+ 
+### LGG, MYB/MYBL1
+
+This subtype harbors either a MYB-QKI fusion or other MYB or MYBL1 fusion
+```{r}
+myb_biospecimen <- lgat_fus %>%
+  filter(MYB_fus=="Yes") %>%
+  mutate(molecular_subtype = "LGG, MYB/MYBL1") %>%
+  select("Kids_First_Biospecimen_ID","molecular_subtype")
+```
+
+### LGG, CDKN2A/B
+
+This subtype harbors focal CDKN2A and/or CDKN2B deletion
+This is a secondary co-occurring alteration with prognostic significance.
+```{r}
+
+cdkn_biospecimen <- lgat_cnv %>%
+  filter(CDKN2A_DEL=="Yes"| CDKN2B_DEL == "Yes") %>%
+  mutate(molecular_subtype = "LGG, CDKN2A/B") %>%
+  select("Kids_First_Biospecimen_ID","molecular_subtype") 
+```
+
+### Compile
+```{r}
+
+all_dna_subtype <- bind_rows(nf1_biospecimen,
+                         brafV600_biospecimen,
+                         otherMAPK_snv_biospecimen,
+                         rtk_snv_biospecimen,
+                         fgfr_cnv_biospecimen,
+                         fgfr_snv_biospecimen,
+                         idh_biospecimen,
+                         h3_biospecimen,
+                         cdkn_biospecimen
+) %>%
+  rename(Kids_First_Biospecimen_ID_DNA=Kids_First_Biospecimen_ID) %>%
+  # add Kids_First_Biospecimen_ID_RNA to dna subtypes dataframe
+  left_join(all_lgat_matched,by="Kids_First_Biospecimen_ID_DNA") 
+
+all_rna_subtype <- bind_rows(kiaabraf_biospecimen,
+                             otherMAPK_fus_biospecimen,
+                             rtk_fus_biospecimen,
+                             fgfr_fus_biospecimen,
+                             myb_biospecimen) %>%
+  rename(Kids_First_Biospecimen_ID_RNA=Kids_First_Biospecimen_ID) %>%
+  # add Kids_First_Biospecimen_ID_DNA to rna subtypes dataframe
+  left_join(all_lgat_matched,by="Kids_First_Biospecimen_ID_RNA")
+
+  
+all_subtype <- bind_rows(all_rna_subtype,
+                         all_dna_subtype) %>%
+  full_join(all_lgat_matched, by=c("Kids_First_Biospecimen_ID_RNA","Kids_First_Biospecimen_ID_DNA",
+                                   "sample_id",
+                                   "Kids_First_Participant_ID")) %>%
+  # lets group because there are some instances where LGAT DNA/RNA have multiple subtypes
+  group_by(sample_id,Kids_First_Participant_ID,Kids_First_Biospecimen_ID_RNA,Kids_First_Biospecimen_ID_DNA) %>%
+  # merging them here with summarise
+  summarise(molecular_subtype= toString(molecular_subtype)) %>%
```

Comment:
Hm, I've never thought about using `summarise()` like this... it's possible [`tidyr::unite()`](https://tidyr.tidyverse.org/reference/unite.html) would give you a bit more control. See other review comment about not including `LGG, `.

Reply:
Oh I see, I was thinking that there were two columns: one with subtype based on RNA data and one with subtype with DNA data, but these are encoded in different _rows_. That's where the `unite()` suggestion was coming from.

Reply:
I did try to do left joins with all_dna_subtypes and all_rna_subtypes and then do `unite()` but seems like the NA's get combined as well then would need to check if `NA, NA` OR `NA, <subtype value>` OR `NA, <subtype value>, <subtype value> etc.. in case_when.

I can work on that part further if using `unite` instead of the 2 step `groupby+summarise` would make the code more clear, let me know!

Reply:
[Newer versions of `tidyr::unite()` have an `na.rm` argument](https://tidyr.tidyverse.org/reference/unite.html), but I _don't think_ the version we have on the project image is new enough (you might double check that if you have not already!).

I don't know that this is worth spending a lot of time on without the `na.rm` functionality.